rust_snuba/src/strategies/clickhouse/batch.rs

```diff
@@ -125,11 +126,14 @@ impl BatchFactory {
         let client = self.client.clone();
 
         let result_handle = self.handle.spawn(async move {
+            let mut sleep_time = 0;
             while receiver.is_empty() && !receiver.is_closed() {
                 // continously check on the receiver stream, only when it's
                 // not empty do we write to clickhouse
+                sleep_time += 800;
                 sleep(Duration::from_millis(800)).await;
             }
+            gauge!("rust_consumer.receiver_sleep_time_ms", sleep_time as u64);
```

Comment:
btw you can avoid the cast to u64 here if you make `sleep_time` a u64 to begin with `let mut sleep_time = 0u64;`